### DISCUSSION TITLE
Vitest unit testing nuxt3

### DISCUSSION THREAD

**Discussion by leunis** [nuxt/nuxt#15974] [Questions] [answered]
I am struggling for a few days now, setting up unit testing for Nuxt3. I am getting errors (https://stackoverflow.com/q/75007823/2118999) when trying to use Vitest. Tried it on macos and ubuntu, same result. And following the documentation (https://nuxt.com/docs/getting-started/testing) using ```@nuxt/test-utils``` gives me the same errors. 
I would love to get it working properly so I can start creating pipelines in GitLab in order to get a proper CI\CD workflow. Can anyone help me. I want to write some good documentation so I can share it with all others that are starting with unit testing and setting up a workflow.
Best regards, Martijn Dekker

**Comment by dsvgl** [comment, marked as answer]
Unit testing is currently under development. You can follow the progress here: https://github.com/nuxt/framework/issues/2465
At the end of this issue a experimental library for testing/feedback was just open-sourced a few days ago: https://github.com/danielroe/vitest-environment-nuxt
Works quite well...

**Reply by leunis** [reply]
Thanks for the directions. Nice that someone already is trying to makte this work. Unfortunately I am getting a command not found for vitest after installing. 
Used https://github.com/nuxt/framework/issues/2465 to make a new comment.
